USAGE.md

```diff
@@ -0,0 +1,112 @@
+# Usage
+
+Usage examples for SendGrid ruby-http-client
+
+## Initialization
+
+```
+require_relative '../lib/ruby_http_client'
+
+# This uses the SendGrid API as an example
+headers = JSON.parse('
+  {
+    "Authorization": "Bearer ' + ENV['SENDGRID_API_KEY'] + '"
+  }
+')
+host = 'https://api.sendgrid.com'
+client = SendGrid::Client.new(host: host, request_headers: headers)
+```
+
+## Table of Contents
+
+- [GET](#get)
+- [DELETE](#delete)
+- [POST](#post)
+- [PUT](#put)
+- [PATCH](#patch)
+
+<a name="get"></a>
+## GET
+
+#### GET Collection
+
+```
+query_params = { 'limit' => 100, 'offset' => 0 }
+response = client.version('v3').api_keys.get(query_params: query_params)
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+#### GET Single
+
+```
+response = client.version('v3').api_keys._(api_key_id).get
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+<a name="delete"></a>
```

Comment:
markdown automatically generates anchors to each header, you can remove all the html tags